### PR TITLE
Regenerate certificates if data has changed

### PR DESCRIPTION
**Pull request by hyperbolic2346**
This makes easyrsa revoke, delete the old certificate, and then generate a new certificate if the data that makes up the certificate has changed.

### CODE REVIEW

**Comment by hyperbolic2346** [comment]
[Allow additional name to be added to certs](https://trello.com/c/LJHAIpfi/214-allow-additional-name-to-be-added-to-certs)

**Review by marcoceppi — COMMENTED** [review]
👏 This 👏 is 👏 what 👏 I'm 👏 talking 👏 about! 🔥🔥🔥

File: reactive/easyrsa.py [hunk 1]
```diff
@@ -253,10 +261,24 @@ def create_server_certificate(cn, san_list, name='server'):
...
-        # Do not regenerate the server certificate if it already exists.
-        if not os.path.isfile(cert_file) and not os.path.isfile(key_file):
-            # Get a string compatible with easyrsa for the subject-alt-names.
-            sans = get_sans(san_list)
+        # Create the path to the request file
+        req_file = 'pki/reqs/{0}.req'.format(name)
+        # Get a string compatible with easyrsa for the subject-alt-names.
+        sans = get_sans(san_list)
+        this_cert = {'sans': sans, 'cn': cn, 'name': name}
+        changed = data_changed('server_cert', this_cert)
```
Comment (Cynerva) [inline reactive/easyrsa.py]: Given that easyrsa creates and manages multiple server certs (see [this loop](https://github.com/hyperbolic2346/layer-easyrsa/blob/7fa2ea21b49f3101c5f74d049cc0d48ca9c560ba/reactive/easyrsa.py#L222-L229)), we probably need the data_changed key to be unique for each certificate, yeah? Something like `data_changed('server_cert.' + name, this_cert)`

Comment (hyperbolic2346) [reply reactive/easyrsa.py]: Yes, good catch!

**Comment by Cynerva** [comment]
Thanks, looks good. Merging!

**Pull request merged**